Simple Inventory Management System/Repositories/ProductRepository.cs

```diff
@@ -0,0 +1,174 @@
+            var parameters = new Dictionary<string, object>
+            {
+                { "@NewName", newName },
+                { "@NewPrice", newPrice },
+                { "@NewQuantity", newQuantity },
+                { "@ProductId", productId.Value }
+            };
+
+            await ExecuteNonQueryAsync(query, parameters, "Product updated successfully.");
+        }
+        else
+        {
+            Console.ForegroundColor = ConsoleColor.Red;
+            Console.WriteLine($"Product with name '{productName}' was not found.");
+            Console.ForegroundColor = ConsoleColor.White;
+        }
+    }
+
+    public async Task DeleteProductAsync(int productId)
+    {
+        try
+        {
+            const string query = "DELETE FROM Products WHERE ProductID = @ProductId";
+            var parameters = new Dictionary<string, object>
+            {
+                { "@ProductId", productId }
+            };
+
+            await ExecuteNonQueryAsync(query, parameters, "Product deleted successfully.");
+        }
+        catch (SqlException ex)
+        {
+            Console.ForegroundColor = ConsoleColor.Red;
+            Console.WriteLine($"Failed to delete product: {ex.Message}");
+            Console.ForegroundColor = ConsoleColor.White;
+        }
+    }
+
+    public async Task SearchProductAsync(string productName)
+    {
+        const string query = "SELECT Name, Price, Quantity FROM Products WHERE Name = @ProductName";
+        var parameters = new Dictionary<string, object>
+        {
+            { "@ProductName", productName }
+        };
+
+        await DisplayProductListAsync(query, parameters);
+    }
+
+    private async Task<int?> GetProductIdAsync(string productName)
+    {
+        const string query = "SELECT ProductID FROM Products WHERE Name = @ProductName";
+        var parameters = new Dictionary<string, object>
+        {
+            { "@ProductName", productName }
+        };
+
+        var reader = await ExecuteReaderAsync(query, parameters);
+        return await reader.ReadAsync() ? (int?)reader["ProductID"] : null;
+    }
+
+    private async Task ExecuteNonQueryAsync(string query, Dictionary<string, object> parameters, string successMessage)
+    {
+        try
+        {
+            await using var connection = new SqlConnection(connectionString);
+            await using var command = new SqlCommand(query, connection);
+            AddParameters(command, parameters);
+            await connection.OpenAsync();
+            await command.ExecuteNonQueryAsync();
+            DisplaySuccessMessage(successMessage);
+        }
+        catch (SqlException ex)
+        {
+            Console.ForegroundColor = ConsoleColor.Red;
+            Console.WriteLine($"Failed operation: {ex.Message}");
+            Console.ForegroundColor = ConsoleColor.White;
+        }
+    }
+
+    private async Task DisplayProductListAsync(string query, Dictionary<string, object>? parameters = null)
```

Comment:
The repo has one task, interact with the database not printing or doing anything else